amaranth/vendor/_xilinx.py

```diff
@@ -1191,29 +1201,56 @@ def get_ff_sync(self, ff_sync):
     def get_async_ff_sync(self, async_ff_sync):
         m = Module()
         m.domains += ClockDomain("async_ff", async_reset=True, local=True)
-        flops = [Signal(1, name=f"stage{index}", reset=1,
-                        attrs={"ASYNC_REG": "TRUE"})
-                 for index in range(async_ff_sync._stages)]
+        # Instantiate a chain of async_ff_sync._stages FDPEs with all
+        # their PRE pins connected to either async_ff_sync.i or
+        # ~async_ff_sync.i. The D of the first FDPE in the chain is
+        # connected to GND.
+        flops_q = Signal(async_ff_sync._stages, reset_less=True)
+        flops_d = Signal(async_ff_sync._stages, reset_less=True)
+        flops_pre = Signal(reset_less=True)
+        for i in range(async_ff_sync._stages):
+            flops = [Instance("FDPE", p_INIT=1, o_Q=flops_q[i],
+                              i_C=ClockSignal(async_ff_sync._o_domain),
+                              i_CE=Const(1), i_PRE=flops_pre, i_D=flops_d[i],
+                              a_ASYNC_REG="TRUE")
+                     for i in range(async_ff_sync._stages)]
```

Comment:
I think the outer `for` isn't doing anything?

Reply:
I think the three `for` loops here should probably be combined into one.

Reply:
Yeah. Silly me. This is a leftover of how I was rewriting the code. I'll fix this.

Reply:
Fixed and force-pushed as a single commit.